docs/rapid-pro-rsdl.md

```diff
@@ -1,92 +1,91 @@
+    name: String
+    incorporated: DateTimeOffset!
 }
 ```
 
-This minimal definition of the Rapid schema contains the following elements:
+This minimal definition of the RAPID schema contains the following elements:
 
-> `type Person`  - definition of the `Person` type
+> `type company`  - definition of the `company` type
 
-> `UserName: String! @RapidID` -
-definition of the UserName field that has `String`
- primitive type and `@RapidID` modifier that assigns it to become Entity ID 
+> `stockSymbol: String! @RapidID` -
+definition of the `stockSymbol` field that has `String`
+primitive type and `@RapidID` directive that assigns it to become Entity ID 
 
+Fields are defined by specifying field name (for example `name`) followed by `:` and a primitive type like `String` etc.
 
-Fields are build by specifying field name (for example `UserName`) followed by `:` and a primitive type like `String` etc.
 
+## Building your service
 
-## Type representation in URI scheme
+A RAPID service consists of top-level resources that can be addressed via URLs.
+These are defined via the special type `Query`:
 
-By default each type is being represented in the URI scheme using it's name.
-Default representation will let developers obtain list of the results (as opposed to single value)
-For example when defining `Person` type in schema as follows :
+```graphql
+type Query {
+    company: company
+    competitors: [company]
+}
+```
 
-> `type Person` => GET `/v4/Rapid/Person`
+Each field of `Query` defines a top-level resource that can be reached via a URI ending in the same name:
 
-Executing `GET` should return the list of the Persons in the database
+> GET `/v4/rapid/company`
 
-### Customization of the URI scheme
+will return the company this service is about.
 
-Developers can control how each type is exposed under URI scheme by utilizing special 
-types `type Query` and `type Mutation`.
+> GET `/v4/rapid/competitors`
 
-`type Query` is being used to expose read operations.
-For example
+will return the list of the competitor companies: `[company]` means that `competitors` is a list resource, and  the list items are of type `company`.
 
-```graphql
-type Query {
-    ## Returns array of person objects (/v4/Rapid/Persons)
-    Persons: [Person]
-    ## Returns single of person object (/v4/Rapid/Admin)
-    Admin: Person
-}
-```
+> GET `/v4/rapid/competitors/cgswl`
 
-By default RapidPro is not exposing operations that can modify resources.
-To enable modifications developers need to specify special `type mutation`
+will return the competitor company identified by the string `cgswl`.
+
+## Enabling resource modification
+
+By default RAPID is not exposing operations that can modify resources.
+These are enabled via the special `type Mutation`:
 
 ```graphql
 type Mutation {
-    ## Allows to perform write operation to the person object
-    Person: Person
+    # Allows update operation to the company resource
+    company: company @RapidUpdate
```

Comment:
This looks strange to me as types are usually starting with upper case.

Reply:
Created #78